File: Algorithms/1342. Number of Steps to Reduce a Number to Zero/javascript/solution.bitwise.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function (num) {
    if (num === 0) {
        return 0;
    }

    let steps = 0,
        powerOfTwo = 1;
    while (powerOfTwo <= num) {
        if (num & powerOfTwo) {
            steps += 2;
        } else {
            steps++;
        }
        powerOfTwo *= 2;
    }
    return steps - 1;
};

File: Algorithms/1342. Number of Steps to Reduce a Number to Zero/golang/solution.bitwise.go
func numberOfSteps(num int) int {
	if num == 0 {
		return 0
	}

	var steps, powerOfTwo int = 0, 1
	for powerOfTwo <= num {
		if num&powerOfTwo > 0 {
			steps++
		}
		steps++
		powerOfTwo *= 2
	}
	return steps - 1
}